core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3984,6 +3985,13 @@ public static List sortArray(List list, boolean ascending) {
     return list;
   }
 
+  /** Support the MAP_CONCAT function. */
+  public static Map mapConcat(Map... maps) {
+    final Map result = new LinkedHashMap();
+    Arrays.stream(maps).forEach(result::putAll);
```

Comment:
I prefer not to use lambda expressions for performance reasons, WDYT?

Reply:
@zoudan i have a question, why  lambda expressions has poor performance

Reply:
I have an impression that lambda expressions have performance issues in certain scenarios in jdk8, but I haven't found clear evidence yet, so it is ok if you want to keep it.

Reply:
@zoudan thanks for your suggestion. i think it makes code clear, so i will not change it